projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -508,990 +324,646 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
-            const sign = (event.deltaY > 0) ? -1 : 1;
-            this.spinValue(event.target.value, sign, event.type);
+    public ngAfterViewChecked() {
+        if (this.labelDirective) {
+            this._renderer.setAttribute(this.inputDirective.nativeElement, 'aria-labelledby', this.labelDirective.id);
         }
     }
 
     /** @hidden @internal */
-    public onInput(event) {
-        /**
-         * Fix for #8165 until refactoring (#6483).
-         * The IgxDateTimeEditor will be used to handle all inputs, i.e. this handler will be removed.
-         * It extends the IgxMaskDirective which contains logic that handles this issue.
-         */
-        if (isIE() && !this._isInEditMode && !this.inputGroup.isFocused) {
-            return;
+    public ngOnDestroy(): void {
+        if (this._overlayId) {
+            this._overlayService.hide(this._overlayId);
         }
-        const targetValue = event.target.value;
-        const cursorPosition = this._getCursorPosition();
-        const checkInput = DatePickerUtil.checkForCompleteDateInput(this.dateFormatParts, targetValue);
-        this._isInEditMode = true;
-
-        if (targetValue !== DatePickerUtil.maskToPromptChars(this.inputMask)) {
-            this.isEmpty = false;
+        if (this._statusChanges$) {
+            this._statusChanges$.unsubscribe();
         }
+        this._destroy$.next();
+        this._destroy$.complete();
+        // TODO: detach()
+    }
 
-        // If all date parts are completed, change the date-picker value, stay in edit mode
-        if (checkInput === 'complete' && event.inputType !== 'deleteContentBackward') {
-            this._transformedDate = targetValue;
-            this.calculateDate(targetValue, event.type);
-            this._setCursorPosition(cursorPosition);
-        } else if (checkInput === 'partial') {
-            // While editing, if one date part is deleted, date-picker value is set to null, the remaining input stays intact.
-            this.deselectDate();
-            requestAnimationFrame(() => {
-                this.getEditElement().value = targetValue;
-                this._setCursorPosition(cursorPosition);
-            });
-        } else if (checkInput === 'empty') {
-            // Total clean-up as input is deleted.
-            this.isEmpty = true;
-            this.deselectDate();
-        }
+    /** @hidden @internal */
+    public getEditElement(): HTMLInputElement {
+        return this.inputDirective.nativeElement;
     }
 
-    public _updateValidityOnBlur() {
-        this._onTouchedCallback();
-        const input = this.inputDirective;
-        if (input && this._ngControl && !this._ngControl.valid) {
-            input.valid = IgxInputState.INVALID;
-        } else {
-            input.valid = IgxInputState.INITIAL;
-        }
+    /** @hidden @internal */
+    public applyCustomFormat() {
+        return this.formatter ? this.formatter(this.value) : this.displayFormat;
```

Comment:
Yeah, those are very much _not_ the same thing. Display format is the template for formatting ('shortDate' or some pattern), while the formatter function straight up returns the formatted value, which is not compatible w/ the display format this is bound to on the directive.
Similar functionality may still be available on the mask/date-time directive, not sure